File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    # 성격 유형 저장 변수
    r=''
    # 4개 성격 유형 지표 리스트 선언
    b=['R','T','C','F','J','M','A','N']
    t=dict(zip(b,[0]*8))
    # 문자열 배열과 선택한 선택지 리스트를 zip함수로 묶고 반복문을 돌면서 성격 유형 지표의 값을 채워줍니다.
    for i,c in zip(survey,choices):
        if c<4:
            t[i[0]]+=4-c
        elif c>4:
            t[i[1]]+=c-4
    # 두 유형별 지표 리스트를 돌면서
    for i in range(0,len(b)//2):
        # 값이 동일하거나 왼쪽 값이 클 때 왼쪽 값 문자열에 추가
        if t[b[2*i]]>=t[b[2*i+1]]:
            r+=b[2*i]
        # 값이 오른쪽 값이 클 때 오른쪽 값 문자열에 추가
        elif t[b[2*i]]<t[b[2*i+1]]:
            r+=b[2*i+1]
    return r

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.Map;
import java.util.HashMap;
class Solution {
    public String solution(String[] survey, int[] choices) {
        String rlt = "";
        char ch;
        Map<Character, Integer> map=new HashMap<>();
        for(int i=0;i<survey.length;i++) {
            ch=survey[i].charAt(choices[i]<4?0:1);
            map.put(ch,map.getOrDefault(ch,0)+(choices[i]<4?1:-1)*(4-choices[i]));
        }
        rlt=new StringBuilder()
            .append(map.getOrDefault('R',0)>=map.getOrDefault('T',0)?'R':'T')
            .append(map.getOrDefault('C',0)>=map.getOrDefault('F',0)?'C':'F')
            .append(map.getOrDefault('J',0)>=map.getOrDefault('M',0)?'J':'M')
            .append(map.getOrDefault('A',0)>=map.getOrDefault('N',0)?'A':'N')
            .toString();
        return rlt;
    }
}